### PR TITLE
fix: avoid unhandled promise rejection in WKSession.send

### PR DESCRIPTION
**Pull request by yury-s**

### CODE REVIEW

File: src/webkit/wkConnection.ts [hunk 1]
```diff
@@ -131,8 +131,12 @@ export class WKSession extends platform.EventEmitter {
     const result = new Promise<Protocol.CommandReturnValues[T]>((resolve, reject) => {
       this._callbacks.set(id, {resolve, reject, error: new Error(), method});
     });
-    this._rawSend(messageObj);
-    return result;
+    try {
+      this._rawSend(messageObj);
+      return result;
+    } catch (e) {
+      return Promise.reject(e);
```
Comment (JoelEinbinder) [inline src/webkit/wkConnection.ts]: I'm trying to wrap my head around this. Previously _rawSend was throwing synchronously. Now we reject and return the rejection. How could something not handling the previous rejection start handling the new one?

Aside from my confusion about what this is doing, I think just marking send as async would do the same thing, without needing the try-catch.

Comment (yury-s) [reply src/webkit/wkConnection.ts]: Done. Changed this to be async.

File: src/webkit/wkExecutionContext.ts [hunk 2]
```diff
@@ -120,21 +120,17 @@ export class WKExecutionContext implements js.ExecutionContextDelegate {
...
+      arguments: serializableArgs.map((arg: any) => this._convertArgument(arg)),
+      returnByValue: false,
+      emulateUserGesture: true
+    }).catch(err => {
       if (err instanceof TypeError && err.message.startsWith('Converting circular structure to JSON'))
         err.message += ' Are you passing a nested JSHandle?';
       throw err;
-    }
-    return callFunctionOnPromise.then(response => {
+    }).then(response => {
```
Comment (aslushnikov) [inline src/webkit/wkExecutionContext.ts]: await's will read better!

Comment (yury-s) [reply src/webkit/wkExecutionContext.ts]: let me take care of that separately

**Review by aslushnikov — APPROVED** [review]

**Pull request merged**